File: app/src/main/java/m13/retrofittest/main/repos/RepositoryDao.java
package m13.retrofittest.main.repos;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface RepositoryDao {
    @Query("SELECT * FROM repositories")
    LiveData<List<Repository>> getAllRepositories();

    @Query("SELECT * FROM repositories WHERE uid IN (:repositoryIds)")
    List<Repository> loadAllByIds(int[] repositoryIds);

    @Query("SELECT * FROM repositories WHERE full_name LIKE :fullName LIMIT 1")
    Repository findByName(String fullName);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Repository... repositories);

    @Delete
    void delete(Repository repository);

    @Query("DELETE FROM repositories")
    void deleteAll();
}

File: app/src/main/java/m13/retrofittest/main/api/repos/RepositoryDao.kt
package m13.retrofittest.main.api.repos

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface RepositoryDao{
    @Query("Select * from Repository")
    fun getAll(): List<Repository>

    @Query("Select * From Repository where uid in (:userIds)")
    fun loadAllByIds(userIds: IntArray):List<Repository>

    /*
     @Query("SELECT * FROM user WHERE first_name LIKE :first AND " +
           "last_name LIKE :last LIMIT 1")
    fun findByName(first: String, last: String): User
     */

    @Insert
    fun insertAll(vararg repositories : Repository)

    @Delete
    fun delete(repository: Repository)
}